library/alloc/src/vec/mod.rs

```diff
@@ -2473,8 +2473,84 @@ impl<T, A: Allocator> Vec<T, A> {
             let end = self.as_mut_ptr().add(self.len);
             ptr::write(end, value);
             self.len += 1;
+            Ok(())
+        }
+    }
+
+    /// Appends an element to the back of a collection, returning a reference to it.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the new capacity exceeds `isize::MAX` _bytes_.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(push_mut)]
+    ///
+    /// let mut vec = vec![];
+    /// // Due to current borrow checker limitations (see -Zpolonius), this is the only way to spell this right now.
+    /// let last = if let Some(v) = vec.last_mut() { v } else { vec.push_mut(0) };
```

Comment:
I'm not sure I understand the point of this.

Reply:
There should probably be a better example, yeah. Will update soon!